Detectors/ZDC/simulation/src/Digits2Raw.cxx

```diff
@@ -591,13 +591,12 @@ void Digits2Raw::emptyBunches(std::bitset<3564>& bunchPattern)
   const int LHCMaxBunches = o2::constants::lhc::LHCMaxBunches;
   mNEmpty = 0;
   for (int32_t ib = 0; ib < LHCMaxBunches; ib++) {
-    int32_t mb = (ib + 31) % LHCMaxBunches; // beam gas from back of calorimeter
-    int32_t m1 = (ib + 1) % LHCMaxBunches;  // previous bunch
-    int32_t cb = ib;                        // current bunch crossing
-    int32_t p1 = (ib - 1) % LHCMaxBunches;  // colliding + 1
-    int32_t p2 = (ib + 1) % LHCMaxBunches;  // colliding + 2
-    int32_t p3 = (ib + 1) % LHCMaxBunches;  // colliding + 3
-    if (bunchPattern[mb] || bunchPattern[m1] || bunchPattern[cb] || bunchPattern[p1] || bunchPattern[p2] || bunchPattern[p3]) {
+    int32_t mb = (ib + 31) % LHCMaxBunches; // beam gas from back of calorimeter (31 bc earlier than a colliding bunch)
+    int32_t m1 = (ib + 1) % LHCMaxBunches;  // next bunch is colliding (position -1)
+    int32_t p1 = (ib - 1) % LHCMaxBunches;  // current bc is 1 bc after colliding (position +1)
+    int32_t p2 = (ib - 2) % LHCMaxBunches;  // current bc is 2 bc after colliding
+    int32_t p3 = (ib - 3) % LHCMaxBunches;  // current bc is 3 bc after colliding
+    if (bunchPattern[mb] || bunchPattern[m1] || bunchPattern[ib] || bunchPattern[p1] || bunchPattern[p2] || bunchPattern[p3]) {
```

Comment:
same problem, you take the % of negative numbers, use `ib + LHCMaxBunches` instead of `ib` in all p.. assignments for prior bunches.

Reply:
Corrected. Thanks